### PR TITLE
Makes items dropped while in a locker go into the locker

### LINKED ISSUE #17230: stripping inside container with full hands puts items on the floor
## Description
it should instead go inside the container that the person being stripped is in
bug can easily blow your cover if you try to loot someone inside a locker or crate, even nobody notices the doafters theyll notice a bunch of shit on the floor at a locker

### PR DESCRIPTION
**Pull request by Mith-randalf**
## About the PR

Fixed unintended behaviour where items dropped while inside a locker went onto the floor, rather than into the locker's storage. Maybe fixes #17230?

## Why / Balance

Bug bad.

## Technical details

Game was trying to insert items into the locker twice. I don't fully understand why it was set up this way, but the second attempt saw the item was already in, freaked out, and dropped it on the floor outside the locker. I just made the first insert attempt be conditional on it not taking place in a container, because there's a second one that (I think) does the same thing if you are. I tested it and it didnt give me any errors and worked. If there's a way this _should_ be working other than this, let me know and I'll change it, the hard part was figuring out what was causing it.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- fix: Due to a budget oversight, items dropped while in lockers were teleporting out of the locker. This has now been rectified.

### CODE REVIEW

**Comment by Plykiya** [comment]
I believe you're not supposed to do RP changelog messages, to make it more clear to users and to shorten the amount they have to read

File: Content.Shared/Hands/EntitySystems/SharedHandsSystem.Drop.cs [hunk 1]
```diff
@@ -109,8 +109,12 @@ public bool TryDrop(EntityUid uid, Hand hand, EntityCoordinates? targetDropLocat
         if (!CanDropHeld(uid, hand, checkActionBlocker))
             return false;
 
+        var userXform = Transform(uid);
+        var isInContainer = ContainerSystem.IsEntityOrParentInContainer(uid, xform: userXform);
+
         var entity = hand.HeldEntity!.Value;
-        DoDrop(uid, hand, doDropInteraction: doDropInteraction, handsComp);
+        if (!isInContainer)
+            DoDrop(uid, hand, doDropInteraction: doDropInteraction, handsComp);
```
Comment (metalgearsloth) [inline Content.Shared/Hands/EntitySystems/SharedHandsSystem.Drop.cs:117]: Isn't this just PlaceNextTo?

**Comment by SoulFN** [comment]
This works with disposal pipes too, this PR fixes it?

File: Content.Shared/Hands/EntitySystems/SharedHandsSystem.Drop.cs (hunk 1 above)
Comment (Mith-randalf) [reply Content.Shared/Hands/EntitySystems/SharedHandsSystem.Drop.cs:117]: Apparently this was caused by a bulk replace, so I need to actually dig through and figure out whats going on better, I just figured DropNextTo was there for a good reason. I'll update.

**Comment by Mith-randalf** [comment]
> This works with disposal pipes too, this PR fixes it?

Yeah, tested it, works as intended.

**Review by ShadowCommander — CHANGES_REQUESTED** [review]
Can you make a test for this? Hands probably needs refactoring and having a test will make this a lot easier.

**Comment by metalgearsloth** [comment]
superceded

**Pull request closed without merging**